src/poetry/installation/chef.py

```diff
@@ -81,7 +83,18 @@ def get_cached_archives_for_link(self, link: Link) -> List[Link]:
         links = []
         for archive_type in archive_types:
             for archive in cache_dir.glob(f"*.{archive_type}"):
-                links.append(Link(archive.as_uri()))
+
+                accept = True
+
+                if link.hash:
+                    real_hash = get_file_hash(archive, link.hash_name)
+                    if real_hash != link.hash:
+                        logger = logging.getLogger(__name__)
+                        logger.warning(f"cache of {link.filename} is corrupted")
```

Comment:
```suggestion
                        logger.warning(f"Cache of {link.filename} is corrupted. It won't be used.")
```
Leaving only a warning that the cache is corrupted, might confuse user.